src/Sprite.js

```diff
@@ -201,6 +201,14 @@ class SpriteBase {
     return ((((deg + 180) % 360) + 360) % 360) - 180;
   }
 
+  // Given a generator function, return a version of it that runs in "warp mode" (no yields).
+  warp(procedure) {
+    return (...args) => {
+      const inst = procedure(...args);
```

Comment:
Just tested this out and realized that we need to bind to `this`.

Using the current `warp` function, this code...
```js
*whenGreenFlagClicked() {
  this.warp(this.spin)();
}

*spin() {
  for (let i = 0; i < 18; i++) {
    this.direction += 10;
    yield;
  }
}
```

...gives this error: `Cannot read property 'direction' of undefined`.

I think the best solution is to modify the warp function like this:
```diff
warp(procedure) {
  return (...args) => {
-   const inst = procedure(...args);
+   const inst = procedure.bind(this)(...args);
    while (!inst.next().done);
  };
}
```

Are there any problems with this solution that I haven't considered?

Reply:
Nope, that looks like it'll work